### PR TITLE
add impl PartialEq for Keypair

### PR DESCRIPTION
**Pull request by rob-solana**
#### Problem
 Keypair doesn't have PartialEq, which complicates things for tests

 #### Summary of Changes
 add impl PartialEq for Keypair, along with From and Deref to make it look like the same type

### CODE REVIEW

**Comment by rob-solana** [comment]
feedback on stupid Rust tricks appreciated

really all I want is for wallet tests to be able to handle a freakin' Keypair in an enum ;)

for reference: https://github.com/dalek-cryptography/ed25519-dalek/pull/82

File: sdk/src/signature.rs [hunk 1]
```diff
@@ -96,7 +127,7 @@ impl KeypairUtil for Keypair {
 pub fn read_keypair(path: &str) -> Result<Keypair, Box<error::Error>> {
     let file = File::open(path.to_string())?;
     let bytes: Vec<u8> = serde_json::from_reader(file)?;
-    let keypair = Keypair::from_bytes(&bytes).unwrap();
+    let keypair = Keypair::from_bytes(&bytes).unwrap(); // TODO: return an error here..
```
Comment (mvines) [inline sdk/src/signature.rs]: ack since you're here could you?

**Comment by garious** [comment]
I'm not opposed to a Solana fork of ed25519-dalek. This isn't the first tweak needed and probably won't be the last.

**Comment by garious** [comment]
@CriesofCarrots, what do you think, spoon ed25519-dalek?

**Comment by CriesofCarrots** [comment]
@garious , I'd be up for that. It would be nice to implement AsRef for Signature as well.

**Comment by mvines** [comment]
My only concern about forking ed25519-dalek would be the optics of it.  Rolling your own crypto is bad mkay and while we would never do that, forking might give somebody the first impression that we have.   Upstream is unresponsive?

**Comment by garious** [comment]
@mvines, I'm not too concerned about optics. I think you're referencing IOTA, but we'd be doing nothing like that. We're not even going as far as a competing implementation of ed25519. Simply a process of getting features upstreamed, but available to ourselves on our timeline, not the upstream's.

**Comment by mvines** [comment]
Changing the description of our fork to something like "Used for upstreaming fixes only" would mostly alleviate my concern about somebody incorrectly coming to the conclusion that we're rolling our own crypto upon seeing our fork.

**Comment by CriesofCarrots** [comment]
I don't see any issue with something like that in the description of our fork. Is that a consensus, then? @garious @mvines 
Or should I hold off until we give it more consideration?

**Comment by mvines** [comment]
wfm, go for it

**Comment by garious** [comment]
Sounds good to me.

**Pull request merged**